Microverse Library Javascript Project

JohnHernCode opened the pull request:
Milestone met:

1. Create HTML/CSS/JS Skeleton
2. Create the book array
3. Display the book array
4. Create a new book form
5. Add button to remove the book
6. Add a button that changes the read status value
7. Store data using localStorage

JohnHernCode committed:
Add base css file

JohnHernCode committed:
Add linters and organize js and css into folders

JohnHernCode committed:
Add Book Function

Commit:
add showBook logic

JohnHernCode committed:
Add Show Books Function

Commit:
create table to show books info

Commit:
add localstorage access

JohnHernCode committed:
Improve localStorage function

Commit:
Add read toggle and remove button

Commit:
Add change read status button

Commit:
add .gitignore

Commit:
Fix linter errors

JohnHernCode committed:
Fix Stylint

JohnHernCode committed:
Fix stylint CSS error

JohnHernCode committed:
Add Readme.md

MahmoudBakr23 reviewed (changes requested):
## **Changes Required 🛑♻**

<hr>

Hello Team :)
I liked your app a lot. You managed some great styling to it, your code is organized, Linter is correct, PR summary, commit messages are descriptive. However, there is always room for improvement. Please consider my comments below 👍 

## **To Improve**

- [x] Please take a quick look at the image below from the project specifications. You should make a new book button to show and hide the books' creation form.
![new_book_button_requirement](https://user-images.githubusercontent.com/47625407/116719010-dd8e7e80-a9da-11eb-9f3d-f6804cc02b09.png)

- [x] Please make sure you don't commit any files or folders that not related to the project. Currently, there are more than 3k changes to review because of the node_modules directory.
<img width="808" alt="gitignore" src="https://user-images.githubusercontent.com/47625407/116719961-f186b000-a9db-11eb-8fd5-d1675f112fe3.png">
Also, as you can notice, you can set up this property globally on your machine to avoid this problem again. 

## **OPTIONAL**

- It'd be great if you include a screenshot of your app in your README file.
- It'd be super nice if you guys managed to deploy this project on a live demo link, then attach it in your README file as well.
- Currently, I can create empty books. So, it'd be great if you put some validations to your form's inputs to prevent that.

<hr>

Good luck and happy coding. 👨‍💻💻
Please feel free to contact me on Slack with the name "Mahmoud Bakr" if you got any questions.

JohnHernCode committed:
removing node_modules

JohnHernCode committed:
Add Collapse to form

Commit:
Add collapsed form js, add screenshot, update readme

Commit:
Update readme

JohnHernCode commented:
Update:

1. Added the new book button for the collapsible form
2. Removed node_modules from the repository
3. Added a screenshot
4. Added a live version
5. Added validations to stop blank books from being added

oloomoses reviewed (approved):
# Congratulations, Project Approved. :muscle: :tada: :clap: 
**Hello team,
Great job working on the suggested changes from the previous reviews.
You have passed all the requirements for this project :muscle: 
Feel free to merge and close this pull request.**

**I wish you the best in your next project and see you at the top :rocket:** 
#### Happy Coding
_Slack: Oloo Moses_
